packages/components/src/components/DownloadButton/DownloadButton.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { Button } from '@mui/material';
+
+export enum DownloadFileType {
+  json = 'application/json',
+  plainText = 'text/plain',
+}
+
+interface IProps {
+  buttonText?: string,
+  fileData: object | string;
+  fileName: string;
+  fileExtension: string;
+  fileType: DownloadFileType;
+}
+
+export const DownloadButton = ({ buttonText = 'Download', fileData, fileName, fileExtension, fileType }: IProps) => {
+  const handleClickDownload = () => {
+    let data = fileData;
+    if (fileType === DownloadFileType.json) {
+      data = JSON.stringify(fileData, null, 2);
+    }
+
+    handleDownloadFile({ fileData: data, fileName, fileExtension, fileType });
+  };
+
+  return (
+    <>
+      <Button onClick={handleClickDownload} variant='contained'>
+        {buttonText}
+      </Button>
+    </>
+  );
+};
+
+/**
+ * handle download file
+ */
+
+export function handleDownloadFile({ fileData, fileName, fileExtension, fileType }: IProps) {
```

Comment:
Should we add this function to utils for reusable?

Reply:
At present, we are exclusively utilizing this function for the Download component.